### PR TITLE
Implemented download test report pdf feature

### PR DESCRIPTION
**Pull request by Suparna-Acharya**
**Fixed Issues**

#22932

**Files Changed**

TestAttendees.cs - Added checkedCandidate Property
package.json -Added jspdf,jspdf-autotable,typings packages
instructions.component.ts-Updated negative sign showing condition
test-report.component.ts-Added downloadtestReportPdf() method and selectALlcandidate() and selectIndividualCandidate() method
test-report.html-Added download button and select checkbox conditions
testAttendee.ts-Added property to client side model
system.config.js-Added path for jspdf package

**Checks**

- [x] Naming Conventions
- [x] Server side code comments (proper English, grammar and no spelling mistakes)
- [x] Client side code comments (proper English, grammar and no spelling mistakes)
- [x] Unused variables, methods, blank spaces and name spaces. 
- [x] Optimal Code and code formatting
- [x] Commit History is proper, no merge is done. 
- [x] Commit/pull request messages should be proper to justify your feature
- [x] All test cases are executed provided by tester.

**Known Issues**

N/A

**Comments**

N/A

---
This change is [Reviewable](https://reviewable.io/reviews/promact/trappist/383)

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/reports/test-report/test-report.component.ts [hunk 1]
```diff
@@ -226,4 +230,67 @@ export class TestReportComponent implements OnInit {
...
+                doc.text(testName, 40, 30);
+            }
+        });
+        doc.save('TestReport.pdf');
+    }
+
+    /**
+     * Select and deselect all candidates of a test
+     */
+    selectAllChandidates() {
```
Comment (rushi216) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/reports/test-report/test-report.component.ts]: Check spelling

File: Trappist/src/Promact.Trappist.Web/wwwroot/systemjs.config.js [hunk 2]
```diff
@@ -51,6 +51,10 @@
             'tinymce': 'npm:tinymce',
             'ngx-popover': 'npm:ngx-popover',
             '@angular/cdk': 'npm:@angular/cdk/bundles/cdk.umd.js'
+            'services': '/services',
```
Comment (rushi216) [inline Trappist/src/Promact.Trappist.Web/wwwroot/systemjs.config.js]: What is the need of this?

File: Trappist/src/Promact.Trappist.Web/package.json [hunk 3]
```diff
@@ -21,13 +21,16 @@
...
+    "jspdf": "^1.3.3",
+    "jspdf-autotable": "^2.3.2",
     "md2": "0.0.18",
     "ng2-ckeditor": "^1.1.6",
     "ngx-clipboard": "5.1.0",
     "ngx-popover": "0.0.16",
     "rxjs": "5.4.2",
     "systemjs": "0.19.40",
     "tinymce": "^4.6.4",
+    "typings": "^2.1.1",
```
Comment (rushi216) [inline Trappist/src/Promact.Trappist.Web/package.json]: We don't use typings library, pls remove

File: Trappist/src/Promact.Trappist.Web/typings.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (rushi216) [inline Trappist/src/Promact.Trappist.Web/typings.json]: same, what is the need of typings if you already have declare let jsPdf: any defined

File: Trappist/src/Promact.Trappist.Web/typings/globals/jspdf/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,206 @@
+// Generated by typings
```
Comment (rushi216) [inline Trappist/src/Promact.Trappist.Web/typings/globals/jspdf/index.d.ts]: Same

File: Trappist/src/Promact.Trappist.Web/typings/globals/jspdf/typings.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (rushi216) [inline Trappist/src/Promact.Trappist.Web/typings/globals/jspdf/typings.json]: Same

File: Trappist/src/Promact.Trappist.Web/typings/index.d.ts [hunk 7]
```diff
@@ -0,0 +1 @@
+/// <reference path="globals/jspdf/index.d.ts" />
```
Comment (rushi216) [inline Trappist/src/Promact.Trappist.Web/typings/index.d.ts]: Same

Comment (rushi216) [inline Trappist/src/Promact.Trappist.Web/typings/index.d.ts]: Remove all typings related files and configuration

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [systemjs.config.js#L5](https://github.com/Promact/trappist/blob/0e3561d10ef2aa6f8a7758e3e8efec76d72e9a4d/Trappist/src/Promact.Trappist.Web/wwwroot/systemjs.config.js#L5): Remove the unused function parameter "global". [rule](http://server12:9005/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [systemjs.config.js#L5](https://github.com/Promact/trappist/blob/0ed974943fad209adb1acb19029b154f3e986b20/Trappist/src/Promact.Trappist.Web/wwwroot/systemjs.config.js#L5): Remove the unused function parameter "global". [rule](http://server12:9005/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**